File: testdata/NestedGenerics.java
import java.util.List;
import java.util.LinkedList;
import java.util.HashSet;

class NestedGenerics
{
  public List< Map<String, ?> > getList() {
    return new LinkedList< Map<String, ?> >(); 
  }
  
  private List< ? extends Set<NestedGenerics> > something = new LinkedList< ? extends Set<NestedGenerics> >();
      
  private Set<List<? super NestedGenerics>> weird = new HashSet<>();      
}

File: testdata/expected/valswithtypes/NestedGenerics.scala
import java.util.List
import java.util.LinkedList
import java.util.HashSet

class NestedGenerics
{
  def getList() : List[ Map[String, _] ] = new LinkedList[ Map[String, _] ]()
  
  private var something : List[ _ <: Set[NestedGenerics] ] = new LinkedList[ _ <: Set[NestedGenerics] ]()
      
  private var weird : Set[List[_ >: NestedGenerics]] = new HashSet()      
}